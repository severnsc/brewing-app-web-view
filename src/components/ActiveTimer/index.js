import React from "react"
import styles from "./styles"
import PropTypes from "prop-types"

const ActiveTimer = ({ name, time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{name}</h2>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	name: PropTypes.string.isRequired,
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer