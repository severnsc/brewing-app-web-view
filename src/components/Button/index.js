import React from "react"
import PropTypes from "prop-types"
import globalStyles from "../styles"

const Button = ({ onClick, value, style }) => (
	<button style={{...globalStyles.button, ...style}} onClick={onClick}>
		{value}
	</button>
)

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number
	]).isRequired,
	style: PropTypes.object
}

export default Button