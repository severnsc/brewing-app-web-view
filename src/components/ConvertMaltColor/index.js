import React from "react"
import PropTypes from "prop-types"
import { MaltColor } from ".."
import { convertColor } from "../../utils"

const ConvertMaltColor = ({ from, to, value }) => {

	const convertedColor = convertColor(from, to, value)

	let srmValue
	if(from === "SRM"){
		srmValue = value
	}else{
		srmValue = convertColor(from, "SRM", value)
	}

	return <MaltColor srmValue={srmValue} value={convertedColor} />

}

ConvertMaltColor.propTypes = {
	from: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	to: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	value: PropTypes.number.isRequired
}

export default ConvertMaltColor