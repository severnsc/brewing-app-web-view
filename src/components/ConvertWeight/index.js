import React from "react"
import PropTypes from "prop-types"
import { Weight } from ".."
import { convertLbsToKg, convertKgToLbs } from "../../utils"

const ConvertWeight = ({ from, to, amount }) => {

	let convertedAmount
	if(from === "imperial" && to === "metric"){
		convertedAmount = convertLbsToKg(amount)
	}

	if(from === "metric" && to === "imperial"){
		convertedAmount = convertKgToLbs(amount)
	}

	return(
		<Weight unit={to} amount={convertedAmount} />
	)
}

ConvertWeight.propTypes = {
	from: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	to: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	amount: PropTypes.number.isRequired
}

export default ConvertWeight