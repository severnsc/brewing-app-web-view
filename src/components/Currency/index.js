import React from "react"
import PropTypes from "prop-types"

const Currency = ({ amount, unit }) => {

	let currencySymbol
	switch(unit){
		case "USD":
			currencySymbol = "$"
			break

		case "GBP":
			currencySymbol = "£"
			break

		case "EUR":
			currencySymbol = "€"
			break

		default:
			currencySymbol = ""
	}

	const parsedAmount = parseFloat(amount).toFixed(2)

	return(
		<span>
			{currencySymbol}{parsedAmount}
		</span>
	)
}

Currency.propTypes = {
	amount: PropTypes.number.isRequired,
	unit: PropTypes.oneOf(["USD", "GBP", "EUR"]).isRequired
}

export default Currency