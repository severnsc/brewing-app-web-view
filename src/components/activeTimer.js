import React from "react"
import PropTypes from "prop-types"

const ActiveTimer = ({ time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const styles = {
		button: {
			color: "white",
			backgroundColor: "grey"
		}
	}

	styles.start = {
		...styles.button,
		backgroundColor: "green"
	}

	styles.stop = {
		...styles.button,
		backgroundColor: "red"
	}

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer