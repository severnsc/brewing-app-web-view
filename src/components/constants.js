export const errorRed = "hsl(353, 85%, 44%)"
export const green = "hsl(167, 90%, 38%)"