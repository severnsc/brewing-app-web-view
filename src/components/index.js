import Timer from "./Timer"
import ForgotPasswordForm from "./ForgotPasswordForm"
import HoverableTableRow from "./HoverableTableRow"
import InventoryItemForm from "./InventoryItemForm"
import Modal from "./Modal"
import NavBar from "./NavBar"
import ResetPasswordForm from "./ResetPasswordForm"
import SearchBar from "./SearchBar"
import SortableTable from "./SortableTable"
import TimerAlertForm from "./TimerAlertForm"
import TimerForm from "./TimerForm"
import InventoryForm from "./InventoryForm"
import FixedNewItemButton from "./FixedNewItemButton"
import NewTimerForm from "./NewTimerForm"
import NewTimerAlertForm from "./NewTimerAlertForm"
import {
	SignupForm,
	LoginForm
} from "./Forms"

export {
	Timer,
	ForgotPasswordForm,
	HoverableTableRow,
	InventoryItemForm,
	LoginForm,
	Modal,
	NavBar,
	ResetPasswordForm,
	SearchBar,
	SignupForm,
	SortableTable,
	TimerAlertForm,
	TimerForm,
	InventoryForm,
	FixedNewItemButton,
	NewTimerForm,
	NewTimerAlertForm
}