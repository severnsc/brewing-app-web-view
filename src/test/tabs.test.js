import React from 'react';
import renderer from "react-test-renderer"
import { Tab, Tabs } from "../components"
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe("Tabs", () => {

	it('renders a tabs component', () => {
		const tabs = renderer.create(
			<Tabs>
				<Tab onClick={() => {}} id="id" active={true} label="tab1" component={<div key="tab1">Tab 1</div>} />
				<Tab onClick={() => {}} active={false} label="tab2" component={<div key="tab2">Tab 2</div>} />
			</Tabs>
		)
		let tree = tabs.toJSON()
		expect(tree).toMatchSnapshot()
	});

	it("renders children", () => {
		const tabs = mount(
			<Tabs>
				<Tab onClick={() => {}} id="id" active={true} label="tab1" component={<div id="tab1" key="tab1">Tab 1</div>} />
				<Tab onClick={() => {}} active={false} label="tab2" component={<div key="tab2">Tab 2</div>} />
			</Tabs>
		)
		expect(tabs.find("#tab1").length).toBe(1)
	})

	it("renders tabs", () => {
		const tabs = shallow(
			<Tabs>
				<Tab onClick={() => {}} active={true} label="tab1" component={<div key="tab1">Tab 1</div>} />
				<Tab onClick={() => {}} active={false} label="tab2" component={<div key="tab2">Tab 2</div>} />
				<Tab onClick={() => {}} active={false} label="tab3" component={<div key="tab3">Tab 3</div>} />
			</Tabs>
		)
		expect(tabs.find(Tab).length).toBe(3)
	})

	it("renders the child of the active tab", () => {
		const tabs = mount(
			<Tabs>
				<Tab onClick={() => {}} active={true} label="tab1" component={<div key="tab1" id="tab1">Tab 1</div>} />
				<Tab onClick={() => {}} active={false} label="tab2" component={<div key="tab2">Tab 2</div>} />
				<Tab onClick={() => {}} active={false} label="tab3" component={<div key="tab3">Tab 3</div>} />
			</Tabs>
		)
		expect(tabs.find("#tab1").length).toBe(1)
	})

	it("does not render children of not active tabs", () => {
		const tabs = mount(
			<Tabs>
				<Tab onClick={() => {}} active={true} label="tab1" component={<div key="tab1" id="tab1">Tab 1</div>} />
				<Tab onClick={() => {}} active={false} label="tab2" component={<div key="tab2" class="inactive">Tab 2</div>} />
				<Tab onClick={() => {}} active={false} label="tab3" component={<div key="tab3" class="inactive">Tab 3</div>} />
			</Tabs>
		)
		expect(tabs.find(".inactive").length).toBe(0)
	})

	it("changes the activeTab onClick", () => {
		const tabs = mount(
			<Tabs>
				<Tab active={true} label="tab1" component={<div key="tab1" id="tab1">Tab 1</div>} />
				<Tab active={false} label="tab2" component={<div key="tab2" className="inactive">Tab 2</div>} />
				<Tab active={false} label="tab3" component={<div key="tab3" className="inactive">Tab 3</div>} />
			</Tabs>
		)
		tabs.find({label: "tab2"}).simulate("click")
		expect(tabs.find(".inactive").length).toBe(1)
	})

})

describe("Tab", () => {

	it("renders a tab component", () => {
		const hello = <div id="hello">Hello</div>
		const tab = renderer.create(
			<Tab onClick={() => {}} active={true} label="label" component={hello}/>
		)
		let tree = tab.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("outputs the label prop", () => {
		const hello = <div id="hello">Hello</div>
		const tab = shallow(
			<Tab onClick={() => {}} active={true} label="label" component={hello} />
		)
		expect(tab.find("li").text()).toBe("label")
	})

	it("calls onClick handler on click", () => {
		const hello = <div id="hello">Hello</div>
		const mockOnClick = jest.fn()
		const tab = shallow(
			<Tab onClick={() => {}} active={true} label="label" component={hello} onClick={mockOnClick} />
		)
		tab.find("li").simulate("click")
		expect(mockOnClick.mock.calls.length).toBe(1)
	})

})