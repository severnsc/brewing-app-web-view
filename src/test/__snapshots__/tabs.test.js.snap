// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders a tab component 1`] = `
<li
  onClick={[Function]}
>
  label
</li>
`;

exports[`Tabs renders a tabs component 1`] = `
<div>
  <ol>
    <li
      onClick={[Function]}
    >
      tab1
    </li>
    <li
      onClick={[Function]}
    >
      tab2
    </li>
  </ol>
  <div>
    <div>
      Tab 1
    </div>
  </div>
</div>
`;
